File: python/SivitasAkademik.py
from Mahasiswa import Mahasiswa

class SivitasAkademik(Mahasiswa):
    def __init__(self, nik, nama, jenis_kelamin, nim, fakultas, prodi, asal_universitas, email_edu):
        super().__init__(nik, nama, jenis_kelamin, nim, fakultas, prodi)
        self.asal_universitas = asal_universitas
        self.email_edu = email_edu
    
    def getFieldValues(self):
        return [self.nik, self.nama, self.jenis_kelamin, self.nim, self.fakultas, self.prodi, self.asal_universitas, self.email_edu]

File: cpp/SivitasAkademik.cpp
#include <iostream>
#include <string>
#include <vector>

using namespace std;

#include "Mahasiswa.cpp"

class SivitasAkademik : public Mahasiswa {
    protected:
        string asal_universitas;
        string email_edu;

    public:

        SivitasAkademik() : Mahasiswa(){

        }

        SivitasAkademik(string NIK, string nama, string jenis_kelamin, string NIM, string fakultas, string prodi, string asal_universitas, string email_edu) : Mahasiswa(NIK, nama, jenis_kelamin, NIM, fakultas, prodi) {
            this->asal_universitas = asal_universitas;
            this->email_edu = email_edu;
        }

        vector<string> getFieldValues() {
            vector<string> values;
            values.push_back(NIK);
            values.push_back(nama);
            values.push_back(jenis_kelamin);
            values.push_back(NIM);
            values.push_back(fakultas);
            values.push_back(prodi);
            values.push_back(asal_universitas);
            values.push_back(email_edu);
            return values;
        }

        ~SivitasAkademik(){}
};